auth-api/src/auth_api/resources/v1/affiliation_invitation.py

```diff
@@ -105,8 +105,9 @@ def post_affiliation_invitation():
 @_jwt.requires_auth
 def get_affiliation_invitation(affiliation_invitation_id):
     """Get the affiliation invitation specified by the provided id."""
-    if not (affiliation_invitation := AffiliationInvitationService.
-            find_affiliation_invitation_by_id(affiliation_invitation_id)):
+    affiliation_invitation = AffiliationInvitationService.find_affiliation_invitation_by_id(affiliation_invitation_id)
+
+    if not affiliation_invitation or affiliation_invitation.as_dict().get('is_deleted'):
```

Comment:
sucks having to do this :(

Reply:
I think it could be avoided by moving the check if something is deleted on the model class.
I didnt test following snippet but I think it could be something along these lines:
```   @classmethod
    def find_invitation_by_id(cls, invitation_id, is_deleted=False):
        """Find an affiliation invitation record that matches the id."""
        return cls.query.filter_by(is_deleted=is_deleted).filter_by(id=invitation_id).first()```
        
It should maintain same behavior as it has now.

Reply:
@hfekete thanks for the feedback, that is a good thought. I will keep it in mind for future similar situations.

In this case I would prefer a find by id method to return the object regardless of its state. I think if we start leveraging the is_deleted flag more for affiliation invitations I would add a new method in the model to filter as per your suggestion. Thanks!